posts/2022-10-27-22.0.0.12-beta.adoc

```diff
@@ -0,0 +1,351 @@
+      
+
+To enable the new Jakarta Server Pages 3.1 and Jakarta Standard Tag Library 3.0 features, add the `pages-3.1` feature to your server.xml:
+[source,xml]
+---- 
+<featureManager>
+   <feature>pages-3.1</feature>
+</featureManager>
+---- 
+ 
+For more information about Jakarta Server Pages and Jakarta Standard Tag Library, refer to the following links:  
+
+* link:{url-prefix}/docs/latest/reference/feature/pages.html[Feature Documentation: Jakarta Server Pages]
+* link:https://jakarta.ee/specifications/pages/3.1/[Specification: Jakarta Server Pages 3.1]
+* link:https://jakarta.ee/specifications/tags/3.0/[Specification: Jakarta Standard Tag Library 3.0]
+* link:https://jakarta.ee/specifications/pages/3.1/apidocs/jakarta.servlet.jsp/module-summary.html[Javadoc: Jakarta Server Pages 3.1]
+* link:https://jakarta.ee/specifications/tags/3.0/apidocs/jakarta.servlet.jsp.jstl/module-summary.html[Javadoc: Jakarta Standard Tag Library 3.0]
+
+// DO NOT MODIFY THIS LINE. </GHA-BLOG-TOPIC> 
+
+
+// // // // DO NOT MODIFY THIS COMMENT BLOCK <GHA-BLOG-TOPIC> // // // // 
+// Blog issue: https://github.com/OpenLiberty/open-liberty/issues/22416
+// Contact/Reviewer: volosied,pnicolucci,ReeceNana
+// // // // // // // // 
+[#websocket]
+== Register endpoints outside of the deployment phase (Jakarta WebSocket 2.1)
+
+Jakarta WebSocket allows communication via the WebSocket protocol, allowing for full-duplex communication between endpoints.  This means that, unlike the HTTP protocol, a single, persisted connection is made across the endpoints, that allows the server to continuously push information to the client. 
+
+Jakarta WebSocket 2.1 is the latest version of the specification and provides several new features and behavior clarifications aimed at improving websocket development. Firstly, when setting up dependencies, developers are likely to notice the API jars have been refactored into client and server jars; however, both APIs are available as part of the `websocket-2.1` feature.
+
+Developers can now register endpoints outside of the deployment phase of the web application, where previously there was a restriction preventing this. HTTP connection upgrades to the websocket protocol can now occur via the link:https://jakarta.ee/specifications/websocket/2.1/apidocs/server/jakarta/websocket/server/servercontainer#upgradeHttpToWebSocket(java.lang.Object,java.lang.Object,jakarta.websocket.server.ServerEndpointConfig,java.util.Map)[`ServerContainer.upgradeHttpToWebSocket()` method​] rather than IBM's specific link:https://www.ibm.com/docs/api/v1/content/SSD28V_liberty/com.ibm.websphere.javadoc.liberty.doc/com.ibm.websphere.appserver.api.wsoc_1.0-javadoc/com/ibm/websphere/wsoc/WsWsocServerContainer.html#doUpgrade(javax.servlet.http.HttpServletRequest,javax.servlet.http.HttpServletResponse,javax.websocket.server.ServerEndpointConfig,java.util.Map)[`WsWsocServerContainer.doUpgrade()` method].

```

Comment:
```suggestion
HTTP connection upgrades to the WebSocket protocol can now occur via the link:https://jakarta.ee/specifications/websocket/2.1/apidocs/server/jakarta/websocket/server/servercontainer#upgradeHttpToWebSocket(java.lang.Object,java.lang.Object,jakarta.websocket.server.ServerEndpointConfig,java.util.Map)[`ServerContainer.upgradeHttpToWebSocket()` method​] rather than IBM's specific link:https://www.ibm.com/docs/api/v1/content/SSD28V_liberty/com.ibm.websphere.javadoc.liberty.doc/com.ibm.websphere.appserver.api.wsoc_1.0-javadoc/com/ibm/websphere/wsoc/WsWsocServerContainer.html#doUpgrade(javax.servlet.http.HttpServletRequest,javax.servlet.http.HttpServletResponse,javax.websocket.server.ServerEndpointConfig,java.util.Map)[`WsWsocServerContainer.doUpgrade()` method].

```